### DISCUSSION TITLE
Good practices of a project

### DISCUSSION THREAD

**Discussion by YuriFontella** [fastify/fastify#3238] [General]
My initial idea, just for fun, was to create a single project using http routes along with graphql in a structure coming close to being clean architecture.

So, i would like to know if this plugin to make queries on bd with knex could be a bad practice?
With him i can use both in graphql resolvers and in the http controller of the routes.

The plugins are in the "infra" folder, and I use them as a dependency on the "app/controller" folder.

[https://github.com/YuriFontella/graphql-rest](https://github.com/YuriFontella/graphql-rest)

**Comment by RafaelGSS** [comment]
What plugin? I think that you forgot to link it.

**Reply by YuriFontella** [reply]
No, it's a plugin that's in the infra folder. It's a plugin with queries, the whole project works very well. But, I don't know if this plugin was a good idea.

[https://github.com/YuriFontella/graphql-rest/blob/main/src/infra/query/index.js](https://github.com/YuriFontella/graphql-rest/blob/main/src/infra/query/index.js)

**Reply by RafaelGSS** [reply]
It's up to you to define. I don't see it as a bad practice, but personally, I wouldn't do that. I likely would use a scoped service for each domain, as a [`Repository` pattern](https://martinfowler.com/eaaCatalog/repository.html). As I said, from a fastify point of view, it's completely ok.

**Reply by YuriFontella** [reply]
The question is not the design pattern or architecture, but that this plugin will grow and have many other queries, and whether this could cause any performance issues with the functions of this plugin being called constantly.

But thank you very much for your reply.

**Reply by RafaelGSS** [reply]
> The question is not the design pattern or architecture, but that this plugin will grow and have many other queries, and whether this could cause any performance issues with the functions of this plugin being called constantly.

No, it wouldn't degrade the app performance.
